Week3/homework/step2-2.js

```diff
@@ -3,21 +3,30 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = 0; i <= stopIndex - startIndex; i++) {
+    numbers.push(startIndex + i);
+    if ((startIndex + i) % 3 === 0 && (startIndex + i) % 5 === 0) {
+      numbers.push(threeCallback(startIndex + i));
+      numbers.push(fiveCallback(startIndex + i));
+    } else if ((startIndex + i) % 3 === 0) {
+      numbers.push(threeCallback(startIndex + i));
+    } else if ((startIndex + i) % 5 === 0) {
+      numbers.push(fiveCallback(startIndex + i));
```

Comment:
👍 for logic. A couple of small suggestions though: 

1. If a number is divisible by both 3 and 5 then that number will also be divisible by 15, isn't it? 😉 Thus, you could also perhaps write the first if condition as `if ((startIndex + i) % 15 === 0) {}`
2. You need not complicate the `for` loop conditions. It could be as simple as `for(let i = startIndex; i <= stopIndex; i++) { // loop body }` 🙂

Reply:
1. Good suggestion. I will change it. 👍 😉
2. I defined "i" just for a loop because of started from 0 and finished (stopIndex- startIndex).  It is the same way but little complicated. I will change thanks